client/client-multi/client-cpp/kaa/KaaClientContext.hpp

```diff
@@ -32,21 +32,23 @@ namespace kaa {
 class KaaClientContext : public IKaaClientContext
 {
 public:
-    KaaClientContext(KaaClientProperties &properties, ILogger &logger,
-                     IExecutorContext &executorContext, IKaaClientStateStoragePtr state = nullptr)
-	  : properties_(properties), logger_(logger), executorContext_(executorContext), state_(state) {}
-
-    virtual KaaClientProperties              &getProperties() { return properties_; }
-    virtual ILogger                              &getLogger() { return logger_; }
-    virtual IKaaClientStateStorage               &getStatus() { return *state_; }
-    virtual IExecutorContext            &getExecutorContext() { return executorContext_; }
+    KaaClientContext(KaaClientProperties &properties, ILogger &logger, IExecutorContext &executorContext,
+        NetworkAddrChangeDetector& detector, IKaaClientStateStoragePtr state = nullptr)
```

Comment:
There are so many changes in tests due to new argument int constructor. How about defaulting it?